From e93ad0d7cddc0a762e6cebfd669273d1ca5f70fd Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Fri, 22 May 2026 16:41:55 +0000
Subject: [PATCH 28/33] drm/sun4i: tcon: Add A523 TCON LCD and TV compatibles

A523 reuses the R40 TCON LCD register layout for TCON_LCD0/1/2 and
the H616 TCON TV layout (pad_sel at BIT(1) of TCON_GCTL) for the
HDMI / eDP TCONs. Bind both new compatibles to the respective quirks.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 drivers/gpu/drm/sun4i/sun4i_tcon.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/sun4i/sun4i_tcon.c b/drivers/gpu/drm/sun4i/sun4i_tcon.c
index 566a44b33..8b2783fc3 100644
--- a/drivers/gpu/drm/sun4i/sun4i_tcon.c
+++ b/drivers/gpu/drm/sun4i/sun4i_tcon.c
@@ -1583,6 +1583,8 @@ const struct of_device_id sun4i_tcon_of_table[] = {
 	{ .compatible = "allwinner,sun20i-d1-tcon-lcd", .data = &sun20i_d1_lcd_quirks },
 	{ .compatible = "allwinner,sun20i-d1-tcon-tv", .data = &sun8i_r40_tv_quirks },
 	{ .compatible = "allwinner,sun50i-h616-tcon-tv", .data = &sun50i_h616_tv_quirks },
+	{ .compatible = "allwinner,sun55i-a523-tcon-lcd", .data = &sun8i_r40_lcd_quirks },
+	{ .compatible = "allwinner,sun55i-a523-tcon-tv", .data = &sun50i_h616_tv_quirks },
 	{ }
 };
 MODULE_DEVICE_TABLE(of, sun4i_tcon_of_table);
-- 
2.53.0

